src/pcre2.h.in

```diff
@@ -496,11 +497,17 @@ For binary compatibility, only add to this list; do not renumber. */
 #define PCRE2_START_OPTIMIZE       68
 #define PCRE2_START_OPTIMIZE_OFF   69
 
-/* Types used in pcre2_set_substitute_case_callout(). */
+/* Types used in pcre2_set_substitute_case_callout().
 
-#define PCRE2_SUBSTITUTE_CASE_LOWER 0
-#define PCRE2_SUBSTITUTE_CASE_UPPER 1
-#define PCRE2_SUBSTITUTE_CASE_TITLE 2
+PCRE2_SUBSTITUTE_CASE_LOWER and PCRE2_SUBSTITUTE_CASE_UPPER are passed to the
+callout to indicate that the case of the entire callout input should be
+case-transformed. PCRE2_SUBSTITUTE_CASE_TITLE_FIRST is passed to indicate that
+only the first character or glyph should be transformed to Unicode titlecase,
+and the rest to lowercase. */
+
+#define PCRE2_SUBSTITUTE_CASE_LOWER        1
+#define PCRE2_SUBSTITUTE_CASE_UPPER        2
+#define PCRE2_SUBSTITUTE_CASE_TITLE_FIRST  3
```

Comment:
should't these have SIMPLE somewhere on their name, so it is clear which kind of transformation is requested?

assume whoever implements the more complicated version will have to use 3 more defines to indicate they do want a "non simple" transformation.

Reply:
These defines are intended for the "non simple" transformation. If all you wanted was simple case transformation, you wouldn't need to set a custom callout.

We're just taking the `\u` and `\l` flags from the replacement string, and telling the callout what was requested by the replacement. The callout can decide to use whatever casing rules it likes to perform that.